source/Platforms.hx

```diff
@@ -0,0 +1,121 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxObject;
+import flixel.FlxSprite;
+import flixel.system.FlxAssets.FlxGraphicAsset;
+import flixel.util.FlxColor;
+import flixel.input.keyboard.FlxKey;
+
+/**
+ * ...
+ * @author Jess Geiger
+ */
+class Platforms extends FlxSprite 
+{
+    //set initial movement
+    private var UP:Bool = false;
+    private var DOWN:Bool = true;
+    private var LEFT:Bool = false;
+    private var RIGHT: Bool = true;
+    //Initialize min and max positioning
+    private var minX:Float;
+    private var maxX:Float;
+    private var minY:Float;
+    private var maxY:Float;
+    //Set starting position and width of platform
+    private var startX:Float;
+    private var startY:Float;
+    private var platformwidth:Int;
+    // private var player:Player;
+    private var tracer:Int = 0;
+    public var offsetX:Float = 0;
+    public var sticky:Bool = false;
+    public var newbool:Bool;
+	public function new(?X:Float=0, ?Y:Float=0, ?W:Int=0, ?L:Float=0, ?R:Float=0, ?U:Float=0, ?D:Float=0, ?trackPlayer:Player, ?SimpleGraphic:FlxGraphicAsset) 
+	{
+		super(X, Y, SimpleGraphic);
+		//set starting positions based on requested dimensions/positions
+        startX = X;
+        startY = Y;
+        minX = L;
+        maxX = R;
+        minY = U;
+        maxY = D;
+        platformwidth = 32 * W;
```

Comment:
Makes a lot of sense, very similar to the pulling gravity out and making it a variable instead of the plain number. The block dimension has been pulled out and made a separate variable, private var blockSize, which has been set at 32, same as the blocks on the map.